src/rocks_record_store.cpp

```diff
@@ -1066,7 +1066,7 @@ namespace mongo {
         _currentSequenceNumber =
           RocksRecoveryUnit::getRocksRecoveryUnit(opCtx)->snapshot()->GetSequenceNumber();
 
-        if (!startIterator.isNull()) {
+        if (!startIterator.isNull() && !_readUntilForOplog.isNull()) {
```

Comment:
startIterator is not-null only in the oplog case. _readUntilForOplog is not-null in oplog case *and* when this is a forward iterator.

So what this fix does is make sure we only use the `startIterator` optimization in forward cursors, and not in backward cursor. Is that what you are trying to do? That seems like a good thing to do, although I don't fully understand the test failures mentioned in #107 to say with confidence that this fix will solve the problem.

Reply:
My intention was to leave this hack for both forward and backward cursors but to remove it for non-oplog capped collections. Frankly speaking, I don't know how this fixes CheckReplOplog - this was just a guess.
After quick code analysis i think `_readUntilForOplog` can be not-null for backward cursors too. Are you sure it cannot?